src/txn/txn_ckpt.c

```diff
@@ -1824,11 +1825,14 @@ __wt_checkpoint_close(WT_SESSION_IMPL *session, bool final)
     }
 
     /*
-     * Don't flush data from trees when there is a stable timestamp set: that can lead to files that
-     * are inconsistent on disk after a crash.
+     * Don't flush data from modified trees independent of system-wide checkpoint if logging is
+     * disabled, and, either there is a stable timestamp set or the connection is configured to
+     * disallow such operation. Flushing trees with logging disabled can lead to files that are
+     * inconsistent on disk after a crash.
```

Comment:
We should avoid the phrase "logging is disabled" in this comment because none of the conditions are looking explicitly at logging. It is just a side effect of the `wt_btree_immediately_durable` call.

Reply:
Fixed